KsApi/models/Reward.swift

```diff
@@ -31,9 +32,19 @@ public struct Reward {
     return self.id == Reward.noReward.id
   }
 
+  /**
+   Returns the closest matching `ShippingRule` for this `Reward` to `otherShippingRule`.
+   If no match is found `otherShippingRule` is returned.
+   */
+  public func shippingRule(matching otherShippingRule: ShippingRule?) -> ShippingRule? {
+    return self.shippingRules?
+      .first { shippingRule in shippingRule.location.id == otherShippingRule?.location.id }
+      ?? otherShippingRule
```

Comment:
Does it make sense to coalesce this to `otherShippingRule`? I'm just thinking if there's no match for some reason that it should use the base reward's shipping cost, but I'm not sure if that would ever happen 🤔

Reply:
I would leave it as nil in the edge case that a shipping rule is not provided. I'm also wondering how we handle creators that specify free shipping? Will that still be a shipping rule?

Reply:
Yeah, let me update to not coalesce this to the matching rule.

Yes, I confirmed it will be returned as a shipping rule with `0` cost.

Reply:
Update comment to explain that this is to work with legacy v1 `Reward`s.